src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```
@@ -0,0 +1,114 @@
+            registrationService.register(new User("validL","123456754",25));
+        });
+    }
+
+    @Test
+    void age_Over18_ok() {
+        Assertions.assertDoesNotThrow(() -> {
+            registrationService.register(new User("validLogin","123456",21));
+            registrationService.register(new User("validL","123456754",100));
+        });
+    }
+
+    @Test
+    void age_Under18_notOk() {
+        Assertions.assertThrows(RegistrationException.class,() -> {
+            registrationService.register(new User("validLogin","12asdga6",0));
+            registrationService.register(new User("validL","123asgda54",-10));
+            registrationService.register(new User("validLog","12dfhsg54",17));
+        });
+    }
+
+    @Test
+    void login_twoUsersWithTheSameLogin_notOk() {
+        User firstUser = new User("sameLogin","1234567",21);
+        User secondUser = new User("sameLogin","1239876",45);
+        Storage.people.add(firstUser);
+
+        RegistrationException actual = Assertions.assertThrows(RegistrationException.class,() -> {
+            registrationService.register(secondUser);
+        });
+        Assertions.assertEquals("User is already added to Storage", actual.getMessage());
+    }
+
+    @Test
+    void password_shortPassword_notOk() {
+        Assertions.assertThrows(RegistrationException.class, () -> {
+            registrationService.register(new User("ValidLogin","short",18));
+            registrationService.register(new User("ValidLogin",null,18));
+            registrationService.register(new User("ValidLogin","s",18));
+        });
+    }
+
+    @Test
+    void login_shortLogin_notOk() {
+        Assertions.assertThrows(RegistrationException.class, () -> {
+            registrationService.register(new User("short","ValidPassword",18));
+            registrationService.register(new User(null,"ValidPassword",18));
+            registrationService.register(new User("s","ValidPassword",18));
```

According to the checklist, you should test edge cases. The minimum login length is 6. Add a test case where the login length is exactly 5, which should throw an exception. You already have tests for logins with length less than 5 and null logins.